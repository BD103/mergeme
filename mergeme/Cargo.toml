[package]
name = "mergeme"
version = "0.1.0"
edition = "2024"

[features]
default = ["derive"]
derive = ["dep:mergeme_derive"]

[dependencies]
mergeme_derive = { path = "../mergeme_derive", version = "0.1.0", optional = true }
