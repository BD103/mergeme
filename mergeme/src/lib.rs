#![no_std]

#[cfg(feature = "derive")]
pub use mergeme_derive::Merge;

pub trait Merge<Partial>: Sized {
    fn merge_in_place(&mut self, other: Partial);

    fn merge(mut self, other: Partial) -> Self {
        self.merge_in_place(other);
        self
    }
}
