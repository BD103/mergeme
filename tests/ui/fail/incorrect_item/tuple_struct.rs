use mergeme::Merge;

#[derive(Merge)]
#[partial(PartialConfig)]
struct Config(bool, u8, Vec<String>);

fn main() {}
