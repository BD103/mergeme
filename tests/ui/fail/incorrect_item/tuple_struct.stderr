error: `#[derive(Merge)]` only works on named structs, not tuple structs
 --> tests/ui/fail/incorrect_item/tuple_struct.rs:4:1
  |
4 | / #[partial(PartialConfig)]
5 | | struct Config(bool, u8, Vec<String>);
  | |_____________________________________^
